impeller/aiks/aiks_unittests.cc

```diff
@@ -219,6 +219,22 @@ TEST_P(AiksTest, CanRenderRadialGradient) {
   ASSERT_TRUE(OpenPlaygroundHere(canvas.EndRecordingAsPicture()));
 }
 
+TEST_P(AiksTest, BlendModeShouldCoverWholeScreen) {
+  Canvas canvas;
+  Paint paint;
+
+  paint.color = Color::Red();
+  canvas.DrawPaint(paint);
+
+  paint.blend_mode = Entity::BlendMode::kSource;
+  canvas.SaveLayer(paint);
```

Comment:
Could you check if this works multiple layers down? I.e. SaveLayer with kSource within a SaveLayer with kSourceOver. 

I expect that this needs to be incorporated in the `EntityPass::GetSubpassCoverage` computation to behave correctly because all of the parent parent layers need to recursively respect this expansion.

Since we always intersect the coverage with the screen size before creating textures, I've been thinking about just having `GetSubpassCoverage` return a coverage rect that stretches to IEEE 754 infinity when the blend mode is destructive.

(As an aside, we also need to do something like this in `Entity::GetCoverage` at some point so that Entities drawn with these destructive blend modes will also result in everything expanding correctly, but doing this infinity trick isn't safe yet since the filter graph has no way to properly limit the size of texture creation yet.)

Reply:
I assume that it works properly.
Skia code https://fiddle.skia.org/c/8c456e807c8a47731ccce2bbc55a48af
skia result 
<img width="266" alt="image" src="https://user-images.githubusercontent.com/54742519/182812864-f1514fcc-eac6-4c3e-bd86-4ce9e39312f7.png">
impeller result
<img width="662" alt="image" src="https://user-images.githubusercontent.com/54742519/182812973-870e99a7-6980-4028-8fa4-246ae42c02e5.png">

Reply:
(If I understand it correctly), I already recursicely handle the `cover_whole_screen_` flag transition between  parrent pass and subpasses in `EntityPass::AddSubpass` and `EntityPass::AddEntity`. It works as long as blend_mode in the subpass doesn't change affer being added to the parrent pass, which is true (if there's a case it changes pls correct me).

Reply:
I am a little bit worried about the IEEE 754 infinity approch in cases where the render target size is determined by coverage of the entity like `RenderToSnapShot`.